runtime/tests/interpreter/if_test.go

```diff
@@ -0,0 +1,300 @@
+/*
+ * Cadence - The resource-oriented smart contract programming language
+ *
+ * Copyright 2019-2020 Dapper Labs, Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package interpreter_test
+
+import (
+	"testing"
+
+	"github.com/stretchr/testify/assert"
+	"github.com/stretchr/testify/require"
+
+	"github.com/onflow/cadence/runtime/interpreter"
+	"github.com/onflow/cadence/runtime/sema"
+	"github.com/onflow/cadence/runtime/tests/checker"
+)
+
+func TestInterpretIfStatement(t *testing.T) {
+
+	t.Parallel()
+
+	inter := parseCheckAndInterpretWithOptions(t,
+		`
+           pub fun testTrue(): Int {
+               if true {
+                   return 2
+               } else {
+                   return 3
+               }
+               return 4
+           }
+
+           pub fun testFalse(): Int {
+               if false {
+                   return 2
+               } else {
+                   return 3
+               }
+               return 4
+           }
+
+           pub fun testNoElse(): Int {
+               if true {
+                   return 2
+               }
+               return 3
+           }
+
+           pub fun testElseIf(): Int {
+               if false {
+                   return 2
+               } else if true {
+                   return 3
+               }
+               return 4
+           }
```

Comment:
```suggestion
           }
           
           pub fun testElseIf(): Int {
               if false {
                   return 2
               } else if false {
                   return 3
               } else {
                   return 4
               }
               return 5
           }
```

adding `else if false`

Reply:
This test is not valid, the name is duplicated and the `return 5` is unreachable